aptos-move/move-examples/fungible_asset/stablecoin/sources/usdk.move

```diff
@@ -0,0 +1,326 @@
+        from: address,
+        from_account_scheme: u8,
+        from_public_key: vector<u8>,
+        to: address,
+        amount: u64,
+    ) acquires Management, State {
+        assert_not_paused();
+        assert_not_denylisted(from);
+        assert_not_denylisted(to);
+
+        let expected_message = Approval {
+            owner: from,
+            nonce: account::get_sequence_number(from),
+            chain_id: chain_id::get(),
+            spender: signer::address_of(spender),
+            amount,
+        };
+        account::verify_signed_message(from, from_account_scheme, from_public_key, proof, expected_message);
+
+        let transfer_ref = &borrow_global<Management>(usdk_address()).transfer_ref;
+        primary_fungible_store::transfer_with_ref(transfer_ref, from, to, amount);
+    }
+
+    /// Deposit function override to ensure that the account is not denylisted and the stablecoin is not paused.
+    public fun deposit<T: key>(
+        store: Object<T>,
+        fa: FungibleAsset,
+        transfer_ref: &TransferRef,
+    ) acquires State {
+        assert_not_paused();
+        assert_not_denylisted(object::owner(store));
+        fungible_asset::deposit_with_ref(transfer_ref, store, fa);
+    }
+
+    /// Withdraw function override to ensure that the account is not denylisted and the stablecoin is not paused.
+    public fun withdraw<T: key>(
+        store: Object<T>,
+        amount: u64,
+        transfer_ref: &TransferRef,
+    ): FungibleAsset acquires State {
+        assert_not_paused();
+        assert_not_denylisted(object::owner(store));
+        fungible_asset::withdraw_with_ref(transfer_ref, store, amount)
+    }
+
+    /// Mint new tokens to the specified account. This checks that the caller is a minter, the stablecoin is not paused,
+    /// and the account is not denylisted.
+    public entry fun mint(minter: &signer, to: address, amount: u64) acquires Management, Roles, State {
+        assert_is_minter(minter);
+        assert_not_paused();
+        assert_not_denylisted(to);
+
+        let management = borrow_global<Management>(usdk_address());
+        let tokens = fungible_asset::mint(&management.mint_ref, amount);
+        deposit(primary_fungible_store::ensure_primary_store_exists(to, metadata()), tokens, &management.transfer_ref);
+
+        event::emit(Mint {
+            minter: signer::address_of(minter),
+            to,
+            amount,
+        });
+    }
+
+    /// Burn tokens from the specified account. This checks that the caller is a minter and the stablecoin is not paused.
+    public entry fun burn(minter: &signer, from: address, amount: u64) acquires Management, Roles, State {
+        burn_from(minter, primary_fungible_store::ensure_primary_store_exists(from, metadata()), amount);
+    }
+
+    /// Burn tokens from the specified account's store. This checks that the caller is a minter and the stablecoin is
+    /// not paused.
+    public entry fun burn_from(
```

Comment:
why are both functions entry?

Reply:
Because they both need to be called from off chain?

Reply:
yes, I meant why is it useful to pass the store object off-chain? more flexibility for defi apps?